components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/edit.jsp

```diff
@@ -239,6 +248,85 @@
                     }
 
                 }
+                function disableAudienceRestriction(chkbx) {
+                    document.editAppform.audience.disabled = (chkbx.checked) ? false
+                        : true;
+                    document.editAppform.addAudience.disabled = (chkbx.checked) ? false
+                        : true;
+                }
+                function addAudienceFunc() {
+                    var audience = $.trim(document.getElementById('audience').value);
+                    if (audience == "") {
+                        document.getElementById("audience").value = "";
+                        return false;
+                    }
+
+                    if ($.inArray(audience, audienceArr) != -1) {
+                        CARBON.showWarningDialog('<fmt:message key="duplicate.audience.value"/>');
+                        document.getElementById("audience").value = "";
+                        return false;
+                    }
+                    audienceArr.push(audience);
+                    var propertyCount = document.getElementById("audiencePropertyCounter");
+
+                    var i = propertyCount.value;
+                    var currentCount = parseInt(i);
+
+                    currentCount = currentCount + 1;
+                    propertyCount.value = currentCount;
+
+                    document.getElementById('audienceTableId').style.display = '';
+                    var audienceTableTBody = document.getElementById('audienceTableTbody');
+
+                    var audienceRow = document.createElement('tr');
+                    audienceRow.setAttribute('id', 'audienceRow' + i);
+
+                    var audience = document.getElementById('audience').value;
+                    var audiencePropertyTD = document.createElement('td');
+                    audiencePropertyTD.setAttribute('style', 'padding-left: 40px ! important; color: rgb(119, 119, 119); font-style: italic;');
+                    audiencePropertyTD.innerHTML = "" + audience + "<input type='hidden' name='audiencePropertyName" + i + "' id='audiencePropertyName" + i + "'  value='" + audience + "'/> ";
```

Comment:
Encode the audience variable to avoid possible XSS.
You can use <%=Encode.forHtml(audience)%>